dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -407,6 +408,90 @@ def new_integration
     render template: Policies::Lti.early_access? ? 'lti/v1/integrations/early_access' : 'lti/v1/integrations'
   end
 
+  # GET /lti/v1/dynamic_registration
+  # Initiate LTI Dynamic Registration flow
+  def new_registration
+    openid_configuration_url = params[:openid_configuration]
+    registration_token = params[:registration_token]
+    begin
+      openid_config = JSON.parse(HTTParty.get(openid_configuration_url).body).symbolize_keys
+    rescue => exception
+      Honeybadger.notify(exception, context: {message: 'Error getting LMS openid_configuration'})
+      return render status: :internal_server_error
+    end
+
+    registration_id = SecureRandom.uuid
+    puts registration_id
+    registration_data = {
+      registration_token: registration_token,
+      registration_endpoint: openid_config[:registration_endpoint],
+      issuer: openid_config[:issuer],
+      registration_id: registration_id,
+    }
+
+    write_cache(registration_id, registration_data, 1.hour)
+    # render form view with any data we'd like to offer as configurable
+    # TEMP placeholder HTML
+    render html: <<-HTML.html_safe
+      <!DOCTYPE html>
+      <html>
+      <head>
+        <title>Simple HTML Page</title>
+      </head>
+      <body>
+        <h1>Hello, World!</h1>
+      </body>
+      </html>
+    HTML
+  end
+
+  # POST /lti/v1/dynamic_registration
+  # Initiate LTI Dynamic Registration Request and create LTI Integration
+  def create_registration
+    admin_email = params[:email]
```

Comment:
TODO:

1. Consider making these params required and...
2. Wrapping more of this code in an exception handler